types/signed_header.go

```diff
@@ -22,7 +26,41 @@ func (sH *SignedHeader) Verify(untrst header.Header) error {
 		// and is a programmer bug
 		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
 	}
+	err := untrstH.ValidateBasic()
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(untrstH.LastHeaderHash[:], sH.Header.Hash()) {
+		return errors.New("Last header hash does not match hash of previous header")
```

Comment:
We should use the go-header Error struct for verification here https://github.com/celestiaorg/go-header/blob/main/errors.go . Also print out the hashes that are not matching for more complete error messages. If this is applicable, please do it for the other errors as well.